webapp/channels/src/components/admin_console/system_properties/user_properties_dot_menu.tsx

```diff
@@ -0,0 +1,181 @@
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {CheckIcon, ChevronRightIcon, DotsHorizontalIcon, EyeOutlineIcon, TrashCanOutlineIcon} from '@mattermost/compass-icons/components';
+import type {FieldVisibility, UserPropertyField} from '@mattermost/types/properties';
+
+import * as Menu from 'components/menu';
+
+import './user_properties_dot_menu.scss';
+import {useUserPropertyFieldDelete} from './user_properties_delete_modal';
+import {isCreatePending} from './user_properties_utils';
+
+type Props = {
+    field: UserPropertyField;
+    updateField: (field: UserPropertyField) => void;
+    deleteField: (id: string) => void;
+}
+
+const menuId = 'user-property-field_dotmenu';
+
+const DotMenu = ({
+    field,
+    updateField,
+    deleteField,
+}: Props) => {
+    const {promptDelete} = useUserPropertyFieldDelete();
+
+    const handleDelete = () => {
+        if (isCreatePending(field)) {
+            // skip prompt when field is pending creation
+            deleteField(field.id);
+        } else {
+            promptDelete(field).then(() => deleteField(field.id));
+        }
+    };
+
+    const handleVisibilityChange = (visibility: FieldVisibility) => {
+        updateField({...field, attrs: {...field.attrs, visibility}});
+    };
+
+    let selectedVisibilityLabel;
+
+    if (field.attrs.visibility === 'always') {
+        selectedVisibilityLabel = (
+            <FormattedMessage
+                id='admin.system_properties.user_properties.dotmenu.visibility.always.label'
+                defaultMessage='Always show'
+            />
+        );
+    } else if (field.attrs.visibility === 'when_set') {
+        selectedVisibilityLabel = (
+            <FormattedMessage
+                id='admin.system_properties.user_properties.dotmenu.visibility.when_set.label'
+                defaultMessage='Hide when empty'
+            />
+        );
+    } else if (field.attrs.visibility === 'hidden') {
+        selectedVisibilityLabel = (
+            <FormattedMessage
+                id='admin.system_properties.user_properties.dotmenu.visibility.hidden.label'
+                defaultMessage='Always hide'
+            />
+        );
+    }
+
+    return (
+        <Menu.Container
+            menuButton={{
+                id: `${menuId}-${field.id}`,
+                class: 'btn btn-transparent user-property-field-dotmenu-menu-button',
+                children: (
+                    <>
+                        <DotsHorizontalIcon size={18}/>
+                    </>
+                ),
+                dataTestId: `${menuId}-${field.id}`,
+                disabled: field.delete_at !== 0,
+            }}
+            menu={{
+                id: `${menuId}-menu`,
+                'aria-label': 'Select an action',
+                className: 'user-property-field-dotmenu-menu',
+            }}
+        >
+            <Menu.SubMenu
```

Comment:
See this comment, https://github.com/mattermost/mattermost/pull/30093#discussion_r1974585972 regarding the menu and accessibility. But I think this same thing will be required for all Menu.Containers.

Reply:
Only necessary with menu containers...all good